src/vs/base/browser/ui/highlightedlabel/highlightedLabel.ts

```diff
@@ -32,7 +32,7 @@ export class HighlightedLabel implements IDisposable {
 		return this.domNode;
 	}
 
-	set(text: string, highlights: IHighlight[] = []) {
+	set(text: string, highlights: IHighlight[] = [], escapedNewLines: boolean = false) {
```

Comment:
I would approach this differently. One method that takes the `text` and `highlights` and that escape them in one go. Something like `static escapeNewLines(text, highlights):[string, IHighligh[]]` and then you call that method early, e.g. before the equality check.